lib/tic_tac_toe/board.ex

```diff
@@ -1,27 +1,87 @@
 defmodule TicTacToe.Board do
   alias TicTacToe.Board
+  alias TicTacToe.BoardInspect
+  alias TicTacToe.BoardUpdate
 
-  defstruct [:board_state, game_over: false]
+  defstruct [:board_state]
 
-  defimpl TicTacToe.BoardInspect do
+  defimpl BoardInspect do
     def get(%Board{board_state: board_state}, position) do
       Enum.at(board_state, position - 1)
     end
+
+    def size(%Board{board_state: board_state}) do
+      number_of_positions = length(board_state)
+      row_length = number_of_positions |> :math.sqrt() |> trunc()
+      {number_of_positions, row_length}
+    end
+
+    def outcome(%Board{board_state: board_state} = board) do
+      winner = winner(board)
+
+      status =
+        cond do
+          winner -> :win
+          draw?(board_state) -> :draw
+          true -> :in_progress
+        end
+
+      {status, winner}
+    end
+
+    defp rows(%Board{board_state: board_state} = board) do
+      {_, row_length} = size(board)
+      1..length(board_state) |> Enum.to_list() |> Enum.chunk_every(row_length)
+    end
+
+    defp columns(rows) do
+      rows |> Enum.zip() |> Enum.map(&Tuple.to_list/1)
+    end
+
+    defp left_diagonal(rows) do
+      rows
+      |> Enum.with_index()
+      |> Enum.map(fn {row, index} -> Enum.at(row, index) end)
+    end
+
+    defp right_diagonal(rows) do
+      Enum.reverse(rows) |> left_diagonal
+    end
+
+    defp winner(%Board{board_state: board_state} = board) do
+      rows = rows(board)
+
+      (rows ++
+         columns(rows) ++ [left_diagonal(rows)] ++ [right_diagonal(rows)])
+      |> Enum.filter(fn [first_position | remaining_positions] ->
+        Enum.all?(remaining_positions, fn x ->
```

Comment:
If you extract a named function here - `winning_row?` maybe, it's a little easier to read.

Reply:
Okay will look into that.  Thanks.

Reply:
I agree. I haven't counted, but I think this might also violate our 5 line constraint.

Reply:
made the change